samples/client/petstore/typescript/tests/default/test/api/PetApi.test.ts

```diff
@@ -120,7 +120,7 @@ describe("PetApi", () =>{
 
     it("uploadFile", (done) => {
         const image = fs.readFileSync(__dirname + "/pet.png")
-        petApi.uploadFile(pet.id, "Metadata", { name: "pet.png", data: image}).then((response: any) => {
+        petApi.uploadFile(pet.id, "Metadata", Object.assign(image, { name: "pet.png" })).then((response: any) => {
```

Comment:
Not sure I agree that mutating the Buffer is more convenient or even more intuitive than passing a composite of {name, data}.
What is the benefit of this approach?

Reply:
True, for node you could argue this not more / less convenient.
I mostly did it because of the browser platform though.
There you usually handle `File` objects, which already contain a `name`.
I'd be happy to hear how you would weigh this.

Reply:
This could also be an generator-specific configuration, couldn't it? Just generate the appropriate wrapper (`{name, data}` composite for anything node, and what is in this PR for the browser) and then internally map this to an appropriate type before doing all the api logic.

Reply:
I could get on board with that, @TiFu. As long as you think that this won't be too much additional complexity.